crates/bevy_ecs/src/schedule/executor_parallel.rs

```diff
@@ -197,6 +197,11 @@ impl ParallelExecutor {
                         .recv()
                         .await
                         .unwrap_or_else(|error| unreachable!(error));
+                    #[cfg(feature = "trace")]
+                    let system_span =
+                        bevy_utils::tracing::info_span!("system", name = &*system.name());
+                    #[cfg(feature = "trace")]
+                    let _system_guard = system_span.enter();
                     unsafe { system.run_unsafe((), world) };
```

Comment:
Should the `_system_guard` be dropped before the await below?

```suggestion
                    #[cfg(feature = "trace")]
                    let system_span =
                        bevy_utils::tracing::info_span!("system", name = &*system.name());
                    #[cfg(feature = "trace")]
                    let system_guard = system_span.enter();
                    unsafe { system.run_unsafe((), world) };
                    #[cfg(feature = "trace")]
                    drop(system_guard);
```

Reply:
the await after is actually instant, but it makes sense